docs/admin/performance/selects.rst

```diff
@@ -107,6 +107,93 @@ to the original result.
    read from the file system. This makes it an ideal candidate for modulo
    based sampling.
 
+
+.. _rewrite-join-as-cte:
+
+Rewrite JOINs as CTEs
+=====================
+
+The article about `using common table expressions to speed up queries`_ shares
+a pattern you can use to replace JOINs with CTEs in your SQL queries, in order
+to achieve consistent and faster execution times.
+
+**Example**
+
+.. code-block:: sql
+
+    -- Uses JOINs
+
+    SELECT SUM(quantity)
+    FROM invoices
+    JOIN invoice_items USING (invoice_number)
+    JOIN products USING (product_id)
+    WHERE product_description='super cool product'
+    AND invoices.issue_date BETWEEN '2024-01-01' AND '2024-02-01';
+
+.. code-block:: sql
+
+    -- Uses CTEs
+
+    WITH relevant_product_ids AS (
+        SELECT product_id
+        FROM products
+        WHERE product_description='super cool product'
+        )
+        ,relevant_invoice_lines AS (
+        SELECT invoice_number,quantity
+        FROM invoice_items
+        WHERE invoice_items.product_id IN (SELECT relevant_product_ids.product_id FROM relevant_product_ids)
+        )
+        ,relevant_invoices AS (
+        SELECT invoice_number,issue_date
+        FROM invoices
+        WHERE invoices.invoice_number IN (SELECT relevant_invoice_lines.invoice_number FROM relevant_invoice_lines)
+        )
```

Comment:
```suggestion
    WITH relevant_product_ids AS (
        SELECT product_id
        FROM products
        WHERE product_description = 'super cool product'
    ),
    relevant_invoice_lines AS (
        SELECT invoice_number, quantity
        FROM invoice_items
        WHERE invoice_items.product_id IN (SELECT relevant_product_ids.product_id FROM relevant_product_ids)
    ),
    relevant_invoices AS (
        SELECT invoice_number, issue_date
        FROM invoices
        WHERE invoices.invoice_number IN (SELECT relevant_invoice_lines.invoice_number FROM relevant_invoice_lines)
    )
```

Although code style is highly subjective, I would argue for two changes here for consistency with other documentation: 1) we don't typically use the "comma on the next line" style, 2) put the closing bracket on the same level of indentation as the matching open one.

Reply:
Hi, and thanks. This style has been inherited from the original source. You may also want to change it over there?

Reply:
Further discussions about SQL (and other) linting and formatting:

- https://github.com/crate/cratedb-guide/pull/45#pullrequestreview-1903183880
- https://github.com/crate/tech-writing/issues/396

Reply:
> 1) we don't typically use the "comma on the next line" style, 2) put the closing bracket on the same level of indentation as the matching open one.

...

> This style has been inherited from the original source. You may also want to change it over there?

I updated the Community post now.